frontend/src/components/profilesettings.js

```diff
@@ -0,0 +1,243 @@
+import React, {Component} from 'react';
+import Avatar from '@material-ui/core/Avatar';
+import Button from '@material-ui/core/Button';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import TextField from '@material-ui/core/TextField';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+import axios from 'axios';
+import { Auth } from '../App'
+
+function getAllAttributes(state, form){
+    let keys = Object.keys(state);
+    keys.forEach((key)=>{
+      state[key] = form[key].value;
+    })
+    return state;
+  }
+
+ 
+const useStyles = makeStyles(theme => ({
+  paper: {
+    marginTop: theme.spacing(8),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.primary.main,
+  },
+  form: {
+    width: '100%', 
+    marginTop: theme.spacing(3),
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+  },
+}));
+
+
+
+export default class UserProfile extends Component {
+   
+  constructor(props){
+    super(props);
+
+    this.onChangeName = this.onChangeName.bind(this);
+    this.onChangeEmail= this.onChangeEmail.bind(this);
+    this.onChangeStreet= this.onChangeStreet.bind(this);
+    this.onChangeCity = this.onChangeCity.bind(this);
+    this.onChangeZip = this.onChangeZip.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+    state = {
+      name: "",
+      email: "",
+      address: {
+          street: "",
+          city: "",
+          zip: ""
+      }
+    }
+    
+    //componentDidMount --> 
+    async componentDidMount() {
+      let self = this;
+      console.log(Auth.isAuthenticated);
+      await axios.get('https://mcgrill-backend.herokuapp.com/user/getUser?email=' + Auth.currentUser, { headers: { Authorization: 'Bearer ' +  Auth.token } })
+      .then(res => res.data)
+      .then(function(resJson) {
+        console.log("json found is:"+resJson);
+      self.setState({
+        name: resJson.name,
+        email: resJson.email,
+        address: resJson.address
+      })})
+      .catch(function (error) {
+        console.log(error);
+    })
+    console.log("state is now:" + this.state.name);
+  }
+  
+    
+  onChangeName(e) {
+    this.setState({
+      name: e.target.value
+    })
+  }
+    
+  onChangeEmail(e) {
+    this.setState({
+      email: e.target.value
+    })
+  }
+
+  onChangeStreet(e) {
+    this.setState({
+      street: e.target.value
+    })
+  }
+   
+  onChangeCity(e) {
+    this.setState({
+      city: e.target.value
+    })
+  }
+
+  onChangeZip(e) {
+    this.setState({
+      zip: e.target.value
+    })
+  }
```

Comment:
these might not work since they exist within the address state (updating nested state is hard). You can make a clone of the address, update it's details, then update the address in state with your modified copy. 

For example in onChangeCity: 

```
let addressCopy = this.state.address;
addressCopy.city = e.target.value;
this.setState({address : addressCopy});

```
and similarly for the others.

Reply:
Thanks for the review @preyansh98 . I've made the suggested changes!